src/go/rpk/pkg/cli/cmd/cluster/config/export.go

```diff
@@ -83,7 +83,7 @@ func exportConfig(
 		if meta.Type == "array" {
 			switch x := curValue.(type) {
 			case nil:
-				fmt.Fprintf(&sb, "%s: []", name)
+				fmt.Fprintf(&sb, "%s:", name)
```

Comment:
Can this be `%s: null`? I think a key with no value technically is not valid yaml.

Reply:
> Both the [node’s properties](https://yaml.org/spec/1.2.2/#node-properties) and [node content](https://yaml.org/spec/1.2.2/#nodes) are optional

Per https://yaml.org/spec/1.2.2/#72-empty-nodes -> Example 7.3. Empty content will be treated as null.

Reply:
We can use null though, but right now we are using empty node content in all the cluster properties, e.g:

```yaml
# Maximum connections per second for one core (may be nil)
kafka_connection_rate_limit:

# Overrides for specific ips for maximum connections per second for one core (e.g. '['127.0.0.1:90', '50.20.1.1:40']')
kafka_connection_rate_limit_overrides: []
# Maximum number of Kafka client connections per broker (may be nil)
kafka_connections_max:

# Per-IP overrides of kafka connection count limit, list of <ip>:<count> strings (e.g. '['127.0.0.1:90', '50.20.1.1:40']')
kafka_connections_max_overrides: []
# Maximum number of Kafka client connections from each IP address, per broker (may be nil)
kafka_connections_max_per_ip:

# Enable authorization for Kafka connections. Values:- `nil`: Ignored. Authorization is enabled with `enable_sasl: true`; `true`: authorization is require    d; `false`: authorization is disabled. See also: `enable_sasl` and `kafka_api[].authentication_method` (may be nil)
kafka_enable_authorization:

# Kafka group recovery timeout expressed in milliseconds
kafka_group_recovery_timeout_ms: 30000

# Principal Mapping Rules for mTLS Authentication on the Kafka API (may be nil)
kafka_mtls_principal_mapping_rules: null
```

Reply:
awesome, I remember looking in the past and thinking empty value was invalid, but since it's valid I'm +1 on it

I'm not sure Go's decoder handles empty though ... but, hopefully that's unrelated :D